editors/code/src/rust-analyzer-api.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * This file mirrors `crates/rust-analyzer/src/req.rs` declarations.
```

Comment:
it's unfortunate that the files are named differently, but I also actively dislike both names :)

Reply:
As an aside I really dislike `req.rs`... it should probably be `custom_requests.rs` or something with only the imports required to support that file. We end up importing a lot from that file in handlers and conv.

Reply:
The name is awesome, unless come up with better!